src/bin/pgcopydb/copydb_schema.c

```diff
@@ -1326,6 +1326,27 @@ copydb_prepare_target_catalog(CopyDataSpec *specs)
 }
 
 
+/*
+ * Convert schema restoreListName to schemaName
+ * The restoreListName is formatted as:
+ * - schema_name owner
+ * We want to extract just the schema_name part
+ */
+bool
+copydb_schema_name_from_restore_list_name(const char *restoreListName,
+										  char *schemaName)
+{
+	if (!(sscanf(restoreListName, "- %64s %*s", schemaName) == 1)) /* IGNORE-BANNED */
```

Comment:
This is a very hopeful way to parse the restore_list_name. When a schema name includes double-quotes, these characters are escaped by being doubled, so we might have up to 128 characters in a schema name. The schema name might contain a space too. The owner name could contain a space also.

Does this archive Table of Contents line contain the schema OID? if yes, we should probably fill-in a SQLite catalog with the list of all the source schemas and then implement an OID lookup in that internal catalog. The way restore_list_name are built is not machine-parsable in the general case.

Reply:
Yeah, I was worrying about it too.

It completely fails to skips anything with a space, this is a toc sample I was looking at from pre-filtered.list.

```
;12; 2615 18530 SCHEMA - aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaabcdefghijklmnop fqayuutgtrkspv
;11; 2615 18529 SCHEMA - aabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghik fqayuutgtrkspv
13; 2615 18531 SCHEMA - abcdefghijklmnopqrstuv wxyz fqayuutgtrkspv
;10; 2615 18528 SCHEMA - abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghik fqayuutgtrkspv
```

There is the OID there, 18530, 18529, 18531.

I presume what you mean is: do a lookup on the source catalog for the oids from s_namespace retrieving the nspame and then use _that_ to lookup the target nspname finally by oid.

Reply:
The lookup should be done in the SQLite catalogs, and for that we need to create and fill-in a new `s_namespace` catalog table in the sourceDB. Once we have that we can lookup the SCHEMA entries by OID in the sourceDB to fetch the schema name (instead of parsing it from the archive contents), and then we can lookup the schema by name in the target database using what you did already.

So yes what you said, with the explicit caching of the whole pg_namespace catalog from the source database in our internal SQLite catalogs.